components/base/kinematicbase/fake_kinematics.go

```diff
@@ -118,14 +118,17 @@ func (fk *fakeDiffDriveKinematics) CurrentPosition(ctx context.Context) (*refere
 
 type fakePTGKinematics struct {
 	*fake.Base
-	localizer   motion.Localizer
-	frame       referenceframe.Frame
-	options     Options
-	sensorNoise spatialmath.Pose
-	origin      *referenceframe.PoseInFrame
-	lock        sync.RWMutex
-	logger      logging.Logger
-	sleepTime   int
+	localizer    motion.Localizer
+	frame        referenceframe.Frame
+	options      Options
+	sensorNoise  spatialmath.Pose
+	ptgs         []tpspace.PTGSolver
```

Comment:
I have changed this driver so that it mimics the `ptgBaseKinematics` driver.
My aim in doing this was to expose to sub-arcs the base executes when executing an input.
Overall, this lends itself in allowing us to confirm that `CheckPlan` can properly interpolate from where the base is to the overall input it was executing (`plan.Trajectory[0]checkFrame.Name()`).

Reply:
oooo, this would be super cool - I will try this out, thanks for the suggestion!

Reply:
ok so I tried doing this but then ran into the following problem. 
When my base moves how does SLAM know it should have moved too?

Reply:
I think we would need a new structure that does not currently exist, which provides both a Base and a Localizer, and when velocity commands are given to the base, will update the localizer's position based on how long it has been at that velocity.

This will be tricky to get right, and is not a small change. Unfortunately, `ptgBaseKinematics`'s only interface with the underlying base is via velocity commands.

Reply:
💭 
🤔